paddle/fluid/operators/mkldnn/fc_mkldnn_op.cc

```diff
@@ -282,6 +295,75 @@ class FCMKLDNNHandler
                                             to_void_cast<float>(bias_data));
   }
 
+  std::shared_ptr<dnnl::memory> AcquireResidualMemory(
+      const phi::DenseTensor* residual, float scale_data) {
+    const std::string residual_key = this->memory_key_ + "@residual";
+    auto memory_p = std::static_pointer_cast<dnnl::memory>(
+        this->dev_ctx_.GetBlob(residual_key));
+    if (!memory_p) {
+      auto dims = this->fwd_pd_->dst_desc().get_dims();
+      if (phi::funcs::is_int8<T_in>()) {
+        auto data_type = residual->dtype() == phi::DataType::INT8
+                             ? dnnl::memory::data_type::s8
+                             : dnnl::memory::data_type::u8;
+
+        auto src_0_md =
+            dnnl::memory::desc(dims, data_type, dnnl::memory::format_tag::ab);
+        auto src_1_md = dnnl::memory::desc(
+            dims, dnnl::memory::data_type::f32, dnnl::memory::format_tag::ab);
+        auto dst_md = dnnl::memory::desc(
+            dims, dnnl::memory::data_type::f32, dnnl::memory::format_tag::ab);
+        std::vector<float> src_data(phi::product(residual->dims()),
+                                    1.f / scale_data);
```

Comment:
can leverage broadcast sementic of binary.
```suggestion
        auto src_1_md = dnnl::memory::desc(
           {1}, dnnl::memory::data_type::f32, dnnl::memory::format_tag::a);
        auto dst_md = dnnl::memory::desc(
           {1}, dnnl::memory::data_type::f32, dnnl::memory::format_tag::a);
        std::vector<float> src_data {1.f / scale_data};
```